docs/mine/lotus/message-pool.md

```diff
@@ -0,0 +1,76 @@
+
+## Gas, fees, limits and caps
+
+When a message is executed it consumes _gas_. The total gas by a message directly affects the cost to place that message in the blockchain, that the sender will have to pay.
+
+In the Filecoin Network, a dynamic **_BaseFee_**, measured in attoFIL/gas units, specifies how much FIL gets burned for every message based on the _gas_ they used for their execution. The _BaseFee_ gets automatically updated according to the network congestion parameters (block sizes). The current value can be obtained from one of the [block explorers](../../get-started/explore-the-network.md) or by looking at the current head:
+
+```sh
+# Will print the last BaseFee in attoFIL
+lotus chain head | xargs lotus chain getblock | jq -r .ParentBaseFee
+```
+
+Apart from the _BaseFee_, every message in the message pool has the following gas-related fields:
+
+- **_GasLimit_**: it is measured in gas units. It a hard limit on the amount of gas that a message's execution can consume. A message consumes gas for every fundamental operation it takes (see [prices](https://github.com/filecoin-project/lotus/blob/d678fe4bfa5b4c70bcebd46cdc38aafc452b42d1/chain/vm/gas.go#L87)). Messages can fail if they runs out of gas fails, at which point every modification to the state is reverted. Whether the execution is successful or not, the miner is given a reward calculated as _GasLimit _ GasPremium\*.
+- **_GasPremium_**: it is measured in attoFIL/gas units. It indicates how much a miner earns as their reward for including a message. A message typically earns its miner _GasLimit \* GasPremium_ attoFIL.
+- **_GasFeeCap_**: it is measured attoFIL/gas units. Its purpose is to establish a hard limit on the amount of FIL a message can cost its sender. A sender is guaranteed that a message will never cost them more than _GasLimit \* GasFeeCap_, not including any value that the message includes for its recipient. The total fee (in attoFIL/gas) for a message is _GasPremium + BaseFee_. Given that _GasPremium_ is set by the sender, _GasFeeCap_ essentially serves as a safeguard against a high, unexpected _BaseFee_.
+
+Note that if a message's _GasFeeCap_ is lower than the _BaseFee_, then the remainder comes from the miner (as a penalty). If _BaseFee + GasPremium_ is less than the message's _GasFeeCap_, then the miner might not earn the entire _GasLimit \* GasPremium_ reward.
+
+## Checking for pending messages in the local pool
+
+If messages are not deemed attractive enough by miners to be included in new blocks, they may become stuck in the local message pool. This is usually a consequence of the _GasFeeCap_ being too low, for example, when the Network's _BaseFee_ is high. It can also be a consequence of the _GasPremium_ being too low if the network is congested.
+
```

Comment:
We should probably mention that at a high level there are w ways to fix stuck messages: (1) change your defaults, including a separate control address and (2) replace a specific message(s).

Reply:
Mentioned them.